src/javascripts/ng-admin/Crud/misc/PromisesResolver.js

```diff
@@ -0,0 +1,53 @@
+/*global define*/
+define(function () {
+    'use strict';
+
+    function PromisesResolver($q) {
+
+        function allEvenFailed(promises) {
+            if (!Array.isArray(promises)) {
+                throw 'allEvenFailed can only handle an array of promises';
+            }
+            var deferred = $q.defer();
+            if (promises.length === 0) {
+                deferred.resolve([]);
+                return deferred.promise;
+            }
+
+            var states = [];
+            var results = [];
+
+            promises.forEach(function (promise, key) {
+                states[key] = false; // promises are not resolved by default
+            });
+
+            promises.forEach(function (promise, key) {
+                function resolve(result) {
+                    states[key] = true;
+                    results[key] = result; // result may be an error
+                    var allFinished = true;
+                    states.forEach(function (state) {
```

Comment:
Can you use a `for` statement to be able to break the loop when we find a `!state` ?